vllm/worker/vineyard_llm_cache.py

```diff
@@ -242,7 +243,7 @@ def prefetch_seq_kv_caches(
         # efficient than performing multiple smaller copy operations. This
         # approach reduces the number of transfers between CPU and GPU,
         # leading to faster overall performance.
-        buffer = self.buffer.cuda()[:, :, :matched]
+        buffer = self.cuda_buffer.copy_(self.buffer)[:, :, :matched]
```

Comment:
What's the purpose of this change? This doesn't seem to be related to unload?

Reply:
this is the optimization for `load`. copying the whole `buffer` to HBM is a contiguous copying and thus more efficient than legacy impl.

Reply:
removing `.cuda()` is because we have introduced `cuda_buffer` for optimizing `unload` and thus we don't need to extra allocation but reusing this `cuda_buffer` for `load` as well. `.cuda()` involves an implicate allocation.

Reply:
So we also applied the contiguous memory copy optimization to load as well? If so, please update the commit/PR description to reflect this.

Also, if Le is only waiting for this memory copy optimization, then I suggest we merge this commit in a separate PR, since it's much smaller than the commit that eliminates the extra re-computations.

Reply:
1. It's already in the description. We have gone through the benefits of the optimization for load in the meeting. The second commit for unload is a follow-up optimization we figured out recently.
2. Actually most of the changes are for load and unload. Changes for skipping re-compute are much smaller and straight forward. I will split the PR later.

Reply:
changes for skipping re-compute have been moved into https://github.com/aibrix/vllm/pull/10

Reply:
Now this PR is self-contained and easier to understand: previously I got the impression that this piece of comments are already there and was wondering why it was not effective and need further tweaks. Now I see all of these changes are part of the optimization of this PR.